crates/turborepo-lib/src/commands/scan.rs

```diff
@@ -0,0 +1,138 @@
+use std::{sync::LazyLock, time::Duration};
+
+use console::{style, Style};
+use indicatif::ProgressBar;
+use turborepo_ui::*;
+
+use super::CommandBase;
+use crate::{diagnostics::*, DaemonPaths};
+
+// can't use LazyCell since DialoguerTheme isn't Sync
+static DIALOGUER_THEME: LazyLock<DialoguerTheme> = LazyLock::new(|| DialoguerTheme {
+    prompt_prefix: style(">>>".to_string()).bright().for_stderr(),
+    active_item_prefix: style("  ❯".to_string()).for_stderr().green(),
+    inactive_item_prefix: style("   ".to_string()).for_stderr(),
+    success_prefix: style("   ".to_string()).for_stderr(),
+    prompt_style: Style::new().for_stderr(),
+    ..Default::default()
+});
+
+/// diagnostics run in parallel however to prevent messages from appearing too
+/// quickly we introduce a minimum delay between each message
+const INTER_MESSAGE_DELAY: Duration = Duration::from_millis(300);
+
+/// Start a diagnostic session. This command will run a series of diagnostics to
+/// help you identify potential performance bottlenecks in your monorepo.
+///
+/// Note: all lints happen in parallel. For the purposes of displaying output,
+///       we demultiplex the output and display it in a single stream, meaning
+///       to the user, it looks like the lints are running serially.
+pub async fn run(base: CommandBase) {
+    let paths = DaemonPaths::from_repo_root(&base.repo_root);
+    let ui = base.ui;
+
+    println!("\n{}\n", ui.rainbow(">>> TURBO SCAN"));
+    println!(
+        "Turborepo does a lot of work behind the scenes to make your monorepo fast,
+however, there are some things you can do to make it even faster. {}\n",
+        BOLD_GREEN.apply_to("Let's go!")
```

Comment:
We should probably obey the color config here
```suggestion
        color!(ui, BOLD_GREEN, "Let's go!")
```